circle.yml

```diff
@@ -1,6 +1,83 @@
+# These environment variables must be set in CircleCI UI
+#
+# DOCKERHUB_REPO - docker hub repo, format: <username>/<repo>
+# DOCKER_EMAIL   - login info for docker hub
+# DOCKER_USER
+# DOCKER_PASS
+
 machine:
+  pre:
+    - curl -sSL https://s3.amazonaws.com/circle-downloads/install-circleci-docker.sh | bash -s -- 1.10.0
   node:
     version: 4.3.1
+  services:
+    - docker
+
 dependencies:
+  cache_directories:
+    - "~/docker"
   pre:
     - npm install -g npm@3
+    - sudo apt-get update; sudo apt-get install pigz
+    - sudo pip install tox mozdownload mozinstall
+    - mozdownload --version latest --destination firefox.tar.bz2
+    - mozinstall firefox.tar.bz2
+    - wget -O geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.10.0/geckodriver-v0.10.0-linux64.tar.gz
```

Comment:
Since we are using Travis for tests today do you think we need to run any tests in CircleCI? I thought it was just a 'nice to have' so that we don't build docker images that fail tests. Maybe just the `npm test` is sufficient here. 

I don't think we can configure downstream build step within CirceCI. I think the only way we can do it is using the 'deployment' section which is not exactly what we want.